### PR TITLE
Added some types. Possibility to pass signOptions

### PR DESCRIPTION
**Pull request by macastro90**

### CODE REVIEW

File: lib/jwtRedisSession.js [hunk 1]
```diff
@@ -1,22 +1,28 @@
...
-const sessionDeserialize = function (o) {
-	return new Promise(function (resolve, reject) {
-		resolve(session);
-	});
+const sessionDeserialize = function (_o) {
+	return Promise.resolve(session);
 };
 
+/**
+ * Create a middleware for JWT Redis Session
```
Comment (reduardo7) [inline lib/jwtRedisSession.js]:
```suggestion
 * Create a middleware for JWT Redis Session.
```

File: lib/jwtRedisSession.js [hunk 2]
```diff
@@ -74,12 +81,12 @@ module.exports = function (options) {
...
 					try {
 						options.sessionDeserialize(JSON.parse(session)).then(function (session) {
-							_.extend(req[options.requestKey], session);
+							extend(req[options.requestKey], session);
 							req[options.requestKey].claims = decoded;
 							req[options.requestKey].id = decoded.jti;
 							req[options.requestKey].jwt = token;
 							// Update the TTL
-							req[options.requestKey].touch(_.noop);
+							req[options.requestKey].touch(noop);
```
Comment (reduardo7) [inline lib/jwtRedisSession.js:91]: Where is defined `noop`?

File: lib/jwtRedisSession.js [hunk 3]
```diff
@@ -1,22 +1,28 @@
-const _ = require("lodash"),
-	jwt = require("jsonwebtoken"),
-	utils = require("./utils");
+const { reduce, extend } = require("lodash");
```
Comment (reduardo7) [inline lib/jwtRedisSession.js]: If you can, use specific _lodash_ imports like following:

https://www.npmjs.com/package/lodash.reduce

File: lib/utils.js [hunk 4]
```diff
@@ -1,34 +1,41 @@
...
+reduce(data, function(memo, val, key){
 	if(typeof val !== "function" && key !== "id")
 		memo[key] = val;
 	return memo;
 }, session);
 };
 
-// these are bound to the session
+/**
+ * These are bound to the session
```
Comment (reduardo7) [inline lib/utils.js]:
```suggestion
 * These are bound to the session.
```

File: lib/utils.js [hunk 5]
```diff
@@ -1,34 +1,41 @@
 
 
-var _ = require("lodash"),
-jwt = require("jsonwebtoken"),
-uuid = require("node-uuid");
+const { reduce, extend } = require("lodash");
```
Comment (reduardo7) [inline lib/utils.js]: Same comment as before.

File: lib/utils.js [hunk 6]
```diff
@@ -1,34 +1,41 @@
```
Comment (reduardo7) [inline lib/utils.js]: Can you remove the blank space?

**Review by reduardo7 — CHANGES_REQUESTED** [review]

File: lib/jwtRedisSession.js (hunk 2 above)
Comment (reduardo7) [reply lib/jwtRedisSession.js:91]: See https://www.npmjs.com/package/lodash.noop

**Comment by reduardo7** [comment]
@macastro90 remember to update the [`README.md`](./README.md) file.

File: lib/jwtRedisSession.js (hunk 3 above)
Comment (macastro90) [reply lib/jwtRedisSession.js]: https://www.npmjs.com/package/lodash.reduce this is a separated package. I couldn't find a package for "extend" lodash utility, so I've kept the "lodash" dependency and I've imported each utility in a separated sentence.

File: lib/jwtRedisSession.js [hunk 7]
```diff
@@ -25,7 +33,7 @@ module.exports = function (options) {
 	options = {
 		client: options.client,
 		secret: options.secret,
-		algorithm: options.algorithm || "HS256",
+		signOptions: options.signOptions,
```
Comment (reduardo7) [inline lib/jwtRedisSession.js:36]: This change should be present in the `README.md` file.

**Review by reduardo7 — CHANGES_REQUESTED** [review]

File: README.md [hunk 8]
```diff
@@ -45,7 +45,9 @@ app.use(JWTRedisSession({
     secret: secret,
     keyspace: "sess:",
     maxAge: 86400,
-    algorithm: "HS256",
+	signOptions: {
+		algorithm: 'HS256'
+	},
```
Comment (reduardo7) [inline README.md]:
```suggestion
    signOptions: {
        algorithm: 'HS256'
    },
```

**Review by reduardo7 — APPROVED** [review]

**Pull request merged**